### PR TITLE
Add support for variable didSet and private(set)

### PR DESCRIPTION
**Pull request by hbmartin**

### CODE REVIEW

**Comment by kdubb** [comment]
Hi @hbmartin thanks for the contribution 🙌. I will go over it in the next few days.

That being said... please don't group unrelated changes into a PR (e.g. your varargs change to `DeclaredTypeName`). It makes generating changes in releases a lot harder.

File: build.gradle.kts [hunk 1]
```diff
@@ -15,7 +15,7 @@ plugins {
 
 
 group = "io.outfoxx"
-version = "1.5.0-SNAPSHOT"
+version = "1.5.1-SNAPSHOT"
```
Comment (kdubb) [inline build.gradle.kts:18]: Please don't commit changes to the version. Versions are handled during releases.

File: src/main/java/io/outfoxx/swiftpoet/ParameterizedTypeName.kt [hunk 2]
```diff
@@ -102,3 +102,6 @@ class ParameterizedTypeName internal constructor(
 
 fun DeclaredTypeName.parameterizedBy(vararg typeArguments: TypeName) =
   ParameterizedTypeName(null, this, typeArguments.toList())
+
```
Comment (kdubb) [inline src/main/java/io/outfoxx/swiftpoet/ParameterizedTypeName.kt:105]: Unrelated change. Please open a new PR. I will say this, that I view this as a pre-optimization. The performance difference here is very small.

**Review by kdubb — COMMENTED** [review]
I've had a cursory look. I need to look more in depth if this is the best way to support this.

**Comment by kdubb** [comment]
@hbmartin Can you make this PR for `didSet` and then open separate ones for `private(set)` and adding list calls?

File: src/test/java/io/outfoxx/swiftpoet/test/FunctionSpecTests.kt [hunk 3]
```diff
@@ -558,6 +558,31 @@ class FunctionSpecTests {
     )
   }
 
+  @Test
+  @DisplayName("Generates parameter modifiers")
+  fun testGenParameterDefault() {
+    val testClass = FunctionSpec.builder("test")
+      .addParameter(
+        ParameterSpec.builder("_ a", INT)
```
Comment (kdubb) [inline src/test/java/io/outfoxx/swiftpoet/test/FunctionSpecTests.kt:566]: You don't create them this way... Use the `ParameterSpec.builder("_", "a", INT)` version.

Comment (hbmartin) [reply src/test/java/io/outfoxx/swiftpoet/test/FunctionSpecTests.kt:566]: Thanks! Also I'll work on splitting this up as soon as I have time. Previous changes were rather urgent for a work project.

**Comment by kdubb** [comment]
PR #79 adds support for `willSet` and `didSet` observers.

Simple usage...
```swift
PropertySpec.varBuilder("test", STRING)
  .didSet(FunctionSpec.willSetBuilder(...))
  .willSet(FunctionSpec.didSetBuilder(...))        
```

**Comment by kdubb** [comment]
Closing PR because these features have been implemented.

**Pull request closed without merging**